core/server/index.js

```diff
@@ -108,6 +112,14 @@ function init(options) {
             });
 
         return new GhostServer(parentApp);
+    }).then(function (_ghostServer) {
+        ghostServer = _ghostServer;
+
+        // scheduling can trigger api requests, that's why we initialize the module after the ghost server creation
+        // scheduling module can create x schedulers with different adapters
+        return scheduling.init(_.extend(config.scheduling, {apiUrl: config.url + config.urlFor('api')}));
+    }).then(function () {
+        return ghostServer;
```

Comment:
It's the index.js one level higher in the route that the npm module users have to re-implement, rather than this one. So it is not a problem. I think using events might be an improvement but one we can revisit later.